### ISSUE TITLE
`UnnecessaryParentheses` false positive with `in` and `?:`

### ISSUE THREAD

**Issue by Gama11** [detekt/detekt#5136]
The following snippet triggers the `UnnecessaryParentheses` rule:

```kt
try {
} catch (e: Exception) {
    if ("foo" in (e.message ?: "")) {}
}
```

However, they're not in fact unnecessary:

![image](https://user-images.githubusercontent.com/2620907/181009684-c2d031d9-3a80-4e35-9531-88436be89bf1.png)

## Your Environment

* Version of detekt used: 1.21.0
* Version of Gradle used (if applicable): 7.5
* Operating System and version: Windows 10 (19044.1766)

**Comment by BraisGabin** [comment]
I can reproduce this with the unit test:

```kotlin
    @ParameterizedTest
    @MethodSource("cases")
    fun `with unnecessary parentheses on aval assignment`(testCase: RuleTestCase) {
        val code = """val a = if ("foo" in (null ?: "")) 1 else 2"""

        assertThat(testCase.rule.lint(code)).hasSize(0)
    }
```

But the issue seems to be in `KtPsiUtil.areParenthesesUseless`. I imagine that this will be solved in the compiler and we will be able to use the fix. But I'm not sure where to report this to be sure that this is fixed.

#4881 would be used as a workaround to this issue if you enable `allowForUnclearPrecedence`. But that's not released yet and as I said, it's just a workaround.

**Comment by BraisGabin** [comment]
This is still an issue right now. But I feel that there is no way to do something about it. Any idea is more than welcome.

But, because this is not actionable, I'm closing it.
